sweep/sweeper.go

```diff
@@ -520,6 +548,11 @@ func (s *UtxoSweeper) collector(blockEpochs <-chan *chainntnfs.BlockEpoch,
 				log.Errorf("schedule sweep: %v", err)
 			}
 
+		// A new external request has been received to retrieve all of
+		// the inputs we're currently attempting to sweep.
+		case req := <-s.pendingSweepsReqs:
+			req.respChan <- s.handlePendingSweepsReq(req)
```

Comment:
I am wondering how this approach compares to just locking the pending inputs while the summary is generated.

Reply:
I think we generally favor this pattern if possible over using a mutex.